src/uct/ib/ud/base/ud_ep.c

```diff
@@ -1196,6 +1202,10 @@ void  uct_ud_ep_disconnect(uct_ep_h tl_ep)
     /* schedule flush */
     uct_ud_ep_flush(tl_ep, 0, NULL);
 
+    /* the EP will be destroyed by interface destroy or timeout in
+     * uct_ud_ep_slow_timer
+     */
+    ep->close_time = ucs_twheel_get_time(&iface->async.slow_timer);
     ep->flags |= UCT_UD_EP_FLAG_DISCONNECTED;
-    /* TODO: at least in debug mode keep and check tl_ep state  */
+    ucs_wtimer_add(&iface->async.slow_timer, &ep->slow_timer, ep->tx.slow_tick);
```

Comment:
maybe need a larger timeout on this one? so will not call the timer so many times before it actually destroys the ep